metashare/api/jobs.py

```diff
@@ -293,3 +298,28 @@ def refresh_github_repositories_for_user(user):
 
 
 refresh_github_repositories_for_user_job = job(refresh_github_repositories_for_user)
+
+
+# This avoids partially-applied saving:
+@transaction.atomic
+def refresh_commits(*, repository, branch_name):
+    """
+    This should only run when we're notified of a force-commit. It's the
+    nuclear option.
+    """
+    from .models import User, Task
+
+    user = User.objects.get(pk=settings.GITHUB_USER_ID)
+    repo = get_repo_info(user, repository.repo_id)
+    commits = repo.commits(repo.branch(branch_name).latest_sha())
+
+    tasks = Task.objects.filter(project__repository=repository, branch_name=branch_name)
+    for task in tasks:
```

Comment:
This _should_ only ever be one task, correct? Not that it hurts anything.

Reply:
Should only be one, yes. But I figured this generalization was a little cleaner.